### PR TITLE
feature flag added

### PR DESCRIPTION
**Pull request by jespy2**
Feature flag for How it works page added.  When disabled, user is sent to landing page.

# References:
This is an update to mntor-2317, which has already been approved but reverted pending this flag
https://github.com/mozilla/blurts-server/pull/4632

# Checklist (Definition of Done)

- [ ] Localization strings (if needed) have been added.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [ ] I've added or updated the relevant sections in readme and/or code comments
- [ ] I've added a unit test to test for potential regressions of this bug.
- [ ] If this PR implements a feature flag or experimentation, the Ship Behind Feature Flag status in Jira has been set
- [ ] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [ ] All acceptance criteria are met.
- [ ] Jira ticket has been updated (if needed) to match changes made during the development process.
- [ ] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: src/app/(proper_react)/(redesign)/(public)/how-it-works/page.tsx [hunk 1]
```diff
@@ -29,7 +30,13 @@ export default async function Page() {
     typeof oneRepActivations === "undefined" ||
     oneRepActivations > monthlySubscribersQuota;
 
-  if (countryCode !== "us") {
+  const featureFlags = await getEnabledFeatureFlags({
+    ignoreAllowlist: true,
+  });
+
+  console.log("featureFlags", featureFlags);
```
Comment (flozia) [inline src/app/(proper_react)/(redesign)/(public)/how-it-works/page.tsx:37]:
```suggestion
```

**Review by flozia — COMMENTED** [review]
We would also need to hide the links to `/how-it-works` if the feature flag is not enabled.  I’m finding those in the following components:
- MobileShell
- Shell
- DashboardTopBannerContent
- WelcomeToPlusView
- Footer

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/DashboardTopBanner/DashboardTopBannerContent.tsx [hunk 2]
```diff
@@ -102,6 +103,10 @@ export const DashboardTopBannerContent = (props: DashboardTopBannerProps) => {
     userDashboardState: UserDashboardState;
     relevantGuidedStep: StepLink;
   }) {
+    console.log(
+      "getDashboardBannerContent-howItWorksFlagEnabled",
+      howItWorksFlagEnabled,
+    );
```
Comment (rhelmer) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/DashboardTopBanner/DashboardTopBannerContent.tsx:109]: Make sure to remove

**Review by rhelmer — COMMENTED** [review]
Thanks for adding this! lgtm but I'll leave approval to front-end folks.

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/DashboardTopBanner/DashboardTopBannerContent.tsx (hunk 2 above)
Comment (jespy2) [reply src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/DashboardTopBanner/DashboardTopBannerContent.tsx]: Done

File: src/app/(proper_react)/(redesign)/(public)/how-it-works/page.tsx (hunk 1 above)
Comment (jespy2) [reply src/app/(proper_react)/(redesign)/(public)/how-it-works/page.tsx]: Updated to remove console logs and add flags

**Review by flozia — APPROVED** [review]

**Pull request merged**